src/language_server_protocol.rs

```diff
@@ -1420,111 +1420,46 @@ impl LanguageClient {
-                        .iter()
-                        .map(|sym| {
-                            let start = sym.location.range.start;
-                            format!(
-                                "{}:{}:\t{}\t\t{:?}",
-                                start.line + 1,
-                                start.character + 1,
-                                sym.name,
-                                sym.kind
-                            )
-                        })
-                        .collect(),
-                    Some(lsp_types::DocumentSymbolResponse::Nested(nested)) => {
-                        let mut symbols = Vec::new();
-
-                        fn walk_document_symbol(
-                            buffer: &mut Vec<String>,
-                            parent: Option<&str>,
-                            ds: &lsp_types::DocumentSymbol,
-                        ) {
-                            let start = ds.selection_range.start;
-
-                            let name = if let Some(parent) = parent {
-                                format!("{}::{}", parent, ds.name)
-                            } else {
-                                ds.name.clone()
-                            };
-
-                            let n = format!(
-                                "{}:{}:\t{}\t\t{:?}",
-                                start.line + 1,
-                                start.character + 1,
-                                name,
-                                ds.kind
-                            );
-
-                            buffer.push(n);
-
-                            if let Some(children) = &ds.children {
-                                for child in children {
-                                    walk_document_symbol(buffer, Some(&ds.name), child);
-                                }
-                            }
-                        }
+        match syms {
+            Some(lsp_types::DocumentSymbolResponse::Flat(flat)) => {
+                self.present_list(&title, &flat)?;
+            }
+            Some(lsp_types::DocumentSymbolResponse::Nested(nested)) => {
+                let mut symbols = Vec::new();
+
+                fn walk_document_symbol(
+                    buffer: &mut Vec<lsp_types::DocumentSymbol>,
+                    parent: Option<&lsp_types::DocumentSymbol>,
+                    ds: &lsp_types::DocumentSymbol,
+                ) {
+                    let name = if let Some(parent) = parent {
+                        format!("{}::{}", parent.name, ds.name)
+                    } else {
+                        ds.name.clone()
+                    };
 
-                        for ds in &nested {
-                            walk_document_symbol(&mut symbols, None, ds);
-                        }
+                    let n = lsp_types::DocumentSymbol {
+                        name: name,
+                        ..ds.clone()
+                    };
```

Comment:
Just a small nit to appease clippy, the name: name thing could be just name. And while we are at it, this whole block could just be:
buffer.push(lsp_types::DocumentSymbol { name, ..ds.clone() });